udev-device.c

```diff
@@ -230,6 +230,23 @@ udev_device_get_sysattr_value(struct udev_device *ud, const char *sysattr)
 	return (NULL);
 }
 
+LIBUDEV_EXPORT int
+udev_device_set_sysattr_value(struct udev_device *ud, const char *sysattr, const char *value)
+{
+	char const *key;
+	struct udev_list_entry *entry;
+
+	udev_list_entry_foreach(entry, udev_list_entry_get_first(&ud->sysattr_list)) {
+		key = _udev_list_entry_get_name(entry);
+		if (!key)
```

Comment:
I'd drop this check in favor of extending the if-statement below to

```
if (key && strcmp(key, sysattr) == 0) return -1;
```

Since there's nothing else in this loop to skip over.